pkg/pubsub/sns.go

```diff
@@ -19,17 +19,34 @@ func NewSNSAdapter(client *sns.SNS) *SNSAdapter {
 	return &SNSAdapter{client}
 }
 
+func (sa *SNSAdapter) parseOpts(opts types.Options) map[string]*sns.MessageAttributeValue {
+	atts := make(map[string]*sns.MessageAttributeValue, 0)
+	for key, val := range opts {
+		atts[key] = &sns.MessageAttributeValue{
```

Comment:
Could do this here as `types.Options`'s value is an interface, you can just check the type and then assign it directly.

```suggestion
		if attributeValue, ok := val.(*sns.MessageAttributeValue) {
			atts[key] = attributeValue
		}
```